packages/extension/src/relay/handlers/content.ts

```diff
@@ -0,0 +1,25 @@
+import { is } from 'superstruct';
+
+import { RelayMessageStruct, RelayResponseStruct, RelayTarget } from '../../types';
+import { createBackgroundMessageSender } from '../background';
+
+export const handleContentMessages = () => {
+  const sendMessage = createBackgroundMessageSender(RelayTarget.Content, RelayTarget.Background);
+
+  window.addEventListener('message', async (event: MessageEvent) => {
+    if (!is(event.data, RelayMessageStruct) || event.data.target !== RelayTarget.Content) {
+      return;
+    }
+
+    const response = await sendMessage(event.data);
+    if (
+      !is(response, RelayResponseStruct) ||
+      response.id !== event.data.id ||
+      response.target !== RelayTarget.Content
+    ) {
+      return;
```

Comment:
Can we test the error state?

Reply:
What do you mean? This is not an error state

Reply:
Sorry, can we test this edge case? I just mean that this line isn't covered 😄